File: C++/issue_21_amisha.cpp
#include <iostream>
using namespace std;

void printStarTriangle(int rows) {
    for (int i = 1; i <= rows; ++i) {
        for (int j = 1; j <= i; ++j) {
            cout << "* ";
        }
        cout << endl;
    }
}

void printNumberTriangle(int rows) {
    for (int i = 1; i <= rows; ++i) {
        for (int j = 1; j <= i; ++j) {
            cout << j << " ";
        }
        cout << endl;
    }
}

void printCharTriangle(int rows) {
    for (int i = 1; i <= rows; ++i) {
        for (int j = 0; j < i; ++j) {
            cout << (char)('A' + j) << " ";
        }
        cout << endl;
    }
}

void printPyramid(int rows) {
    for (int i = 1; i <= rows; ++i) {
        for (int j = 1; j <= rows - i; ++j) {
            cout << " ";
        }
        for (int k = 1; k <= (2 * i - 1); ++k) {
            cout << "*";
        }
        cout << endl;
    }
}

void printDiamond(int rows) {
    for (int i = 1; i <= rows; ++i) {
        for (int j = 1; j <= rows - i; ++j) {
            cout << " ";
        }
        for (int k = 1; k <= (2 * i - 1); ++k) {
            cout << "*";
        }
        cout << endl;
    }
    for (int i = rows - 1; i >= 1; --i) {
        for (int j = 1; j <= rows - i; ++j) {
            cout << " ";
        }
        for (int k = 1; k <= (2 * i - 1); ++k) {
            cout << "*";
        }
        cout << endl;
    }
}

void printHollowSquare(int size) {
    for (int i = 1; i <= size; ++i) {
        for (int j = 1; j <= size; ++j) {
            if (i == 1 || i == size || j == 1 || j == size) {
                cout << "* ";
            } else {
                cout << "  "; 
            }
        }
        cout << endl;
    }
}

int main() {
    int choice;
    int rows;

    cout << "Choose a pattern to print:" << endl;
    cout << "1. Star Triangle" << endl;
    cout << "2. Number Triangle" << endl;
    cout << "3. Character Triangle" << endl;
    cout << "4. Pyramid" << endl;
    cout << "5. Diamond" << endl;
    cout << "6. Hollow Square" << endl;
    cout << "Enter your choice (1-6): ";
    cin >> choice;

    cout << "Enter the number of rows/size: ";
    cin >> rows;

    switch (choice) {
        case 1:
            printStarTriangle(rows);
            break;
        case 2:
            printNumberTriangle(rows);
            break;
        case 3:
            printCharTriangle(rows);
            break;
        case 4:
            printPyramid(rows);
            break;
        case 5:
            printDiamond(rows);
            break;
        case 6:
            printHollowSquare(rows);
            break;
        default:
            std::cout << "Invalid choice." << std::endl;
            break;
    }

    return 0;
}

File: JavaScript/issue_21_amisha.js
function printPattern() {
    const patternType = prompt("Which pattern do you want to print? Choose from: stars, numbers, characters, pyramid, diamond, triangle, hollow: ").toLowerCase();
    const rows = parseInt(prompt("Enter the number of rows for the pattern: "));

    if (isNaN(rows) || rows <= 0) {
        console.log("Invalid number of rows. Please enter a positive integer.");
        return;
    }

    switch (patternType) {
        case "stars":
            for (let i = 1; i <= rows; i++) {
                console.log("* ".repeat(i));
            }
            break;
        case "numbers":
            for (let i = 1; i <= rows; i++) {
                let row = "";
                for (let j = 1; j <= i; j++) {
                    row += j + " ";
                }
                console.log(row);
            }
            break;
        case "characters":
            for (let i = 0; i < rows; i++) {
                let row = "";
                for (let j = 0; j <= i; j++) {
                    row += String.fromCharCode(65 + j) + " "; 
                }
                console.log(row);
            }
            break;
        case "pyramid":
            for (let i = 1; i <= rows; i++) {
                let row = " ".repeat(rows - i);
                row += "* ".repeat(i);
                console.log(row);
            }
            break;
        case "diamond":
            // Upper part of the diamond
            for (let i = 1; i <= rows; i++) {
                let row = " ".repeat(rows - i);
                row += "* ".repeat(i);
                console.log(row);
            }
            // Lower part of the diamond
            for (let i = rows - 1; i >= 1; i--) {
                let row = " ".repeat(rows - i);
                row += "* ".repeat(i);
                console.log(row);
            }
            break;
        case "triangle": 
            for (let i = 1; i <= rows; i++) {
                console.log("* ".repeat(i));
            }
            break;
        case "hollow":
            for (let i = 1; i <= rows; i++) {
                let row = "";
                if (i === 1 || i === rows) {
                    row = "* ".repeat(rows);
                } else {
                    row = "* " + " ".repeat(2 * (rows - 2)) + "*";
                }
                console.log(row);
            }
            break;
        default:
            console.log("Invalid pattern type selected.");
    }
}

printPattern();
